app/src/app/components/sidebar/PaintByCounty.tsx

```diff
@@ -0,0 +1,42 @@
+import { Text, Checkbox, Flex } from "@radix-ui/themes";
+import { useMapStore } from "@/app/store/mapStore";
+import { COUNTY_LAYER_IDS } from "../../constants/layers";
+import { useState } from "react";
+
+export default function PaintByCounty() {
+  const { mapRef, addVisibleLayerIds } = useMapStore((state) => ({
+    mapRef: state.mapRef,
+    addVisibleLayerIds: state.addVisibleLayerIds,
+  }));
+  const [checked, setChecked] = useState(false);
+
+  const paintByCounty = (prevIsChecked: boolean) => {
+    console.log("paintByCounty", prevIsChecked);
+    if (!mapRef || !mapRef?.current) return;
+    if (!prevIsChecked) {
+      console.log("BOX WAS NOT CHECKED, ADDING COUNTIES AS VISIBLE LAYERS");
+      COUNTY_LAYER_IDS.forEach((layerId) => {
+        mapRef.current?.setLayoutProperty(layerId, "visibility", "visible");
```

Comment:
I left some other comments in `mapStore.ts`, but I get an error where there's some rendering confusion between here and the `Map` object: 
<img width="1388" alt="image" src="https://github.com/user-attachments/assets/7f7b70d8-65d0-4891-877b-3c9883b19dcd">
I put a full stack trace at 140 in `mapStore.ts`

Reply:
It's worth investigating a bit more but I think we will also want to factor out some `useEffects` in `Map` regardless, but there may be a re-render happening there when I toggle this

Reply:
Just was a bad implementation in PaintByCounty. I was attempting to modify state within a component and outside a useEffect 🙃